gst/nnstreamer/tensor_query/tensor_query_serversrc.c

```diff
@@ -292,97 +442,20 @@ gst_tensor_query_serversrc_get_property (GObject * object, guint prop_id,
   }
 }
 
-/**
- * @brief nnstreamer-edge event callback.
- */
-static int
-_nns_edge_event_cb (nns_edge_event_h event_h, void *user_data)
-{
-  nns_edge_event_e event_type;
-  int ret = NNS_EDGE_ERROR_NONE;
-
-  GstTensorQueryServerSrc *src = (GstTensorQueryServerSrc *) user_data;
-  if (0 != nns_edge_event_get_type (event_h, &event_type)) {
-    nns_loge ("Failed to get event type!");
-    return NNS_EDGE_ERROR_UNKNOWN;
-  }
-
-  switch (event_type) {
-    case NNS_EDGE_EVENT_NEW_DATA_RECEIVED:
-    {
-      nns_edge_data_h data;
-
-      nns_edge_event_parse_new_data (event_h, &data);
-      g_async_queue_push (src->msg_queue, data);
-      break;
-    }
-    default:
-      break;
-  }
-
-  return ret;
-}
-
-/**
- * @brief start processing of query_serversrc, setting up the server
- */
-static gboolean
-gst_tensor_query_serversrc_start (GstBaseSrc * bsrc)
-{
-  GstTensorQueryServerSrc *src = GST_TENSOR_QUERY_SERVERSRC (bsrc);
-  char *id_str = NULL, *port = NULL;
-
-  id_str = g_strdup_printf ("%d", src->src_id);
-  src->server_h = gst_tensor_query_server_add_data (id_str, src->connect_type);
-  g_free (id_str);
-
-  src->edge_h = gst_tensor_query_server_get_edge_handle (src->server_h);
-  if (src->host)
-    nns_edge_set_info (src->edge_h, "HOST", src->host);
-  if (src->port > 0) {
-    port = g_strdup_printf ("%u", src->port);
-    nns_edge_set_info (src->edge_h, "PORT", port);
-    g_free (port);
-  }
-  if (src->dest_host)
-    nns_edge_set_info (src->edge_h, "DEST_HOST", src->dest_host);
-  if (src->dest_port > 0) {
-    port = g_strdup_printf ("%u", src->dest_port);
-    nns_edge_set_info (src->edge_h, "DEST_PORT", port);
-    g_free (port);
-  }
-  if (src->topic)
-    nns_edge_set_info (src->edge_h, "TOPIC", src->topic);
-
-  nns_edge_set_event_callback (src->edge_h, _nns_edge_event_cb, src);
-
-  if (NNS_EDGE_ERROR_NONE != nns_edge_start (src->edge_h)) {
-    nns_loge
-        ("Failed to start NNStreamer-edge. Please check server IP and port.");
-    return FALSE;
-  }
-
-  if (!gst_tensor_query_server_wait_sink (src->server_h)) {
-    nns_loge ("Failed to get server information from query server.");
-    return FALSE;
-  }
-
-  return TRUE;
-}
-
 /**
  * @brief Get buffer from message queue.
  */
 static GstBuffer *
 _gst_tensor_query_serversrc_get_buffer (GstTensorQueryServerSrc * src)
 {
-  nns_edge_data_h data_h;
+  nns_edge_data_h data_h = NULL;
   GstBuffer *buffer = NULL;
   guint i, num_data;
   GstMetaQuery *meta_query;
   int ret;
 
-  data_h = g_async_queue_pop (src->msg_queue);
+  while (src->playing && !data_h)
+    data_h = g_async_queue_timeout_pop (src->msg_queue, 100000U);
```

Comment:
It would be better to make a const variable for ```100000U```.

Reply:
Updated.